reward/stakingInfo.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package reward
+
+import (
+	"errors"
+	"fmt"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/params"
+	"math"
+	"math/big"
+	"sort"
+)
+
+const (
+	AddrNotFoundInCouncilNodes = -1
+	maxStakingLimit            = uint64(100000000000)
+	DefaultGiniCoefficient     = -1.0
+)
+
+var (
+	maxStakingLimitBigInt = big.NewInt(0).SetUint64(maxStakingLimit)
+
+	ErrAddrNotInStakingInfo = errors.New("Address is not in stakingInfo")
+)
+
+// StakingInfo contains staking information.
+type StakingInfo struct {
+	BlockNum uint64 // Block number where staking information of Council is fetched
+
+	// Information retrieved from AddressBook smart contract
+	CouncilNodeAddrs    []common.Address // NodeIds of Council
+	CouncilStakingAddrs []common.Address // Address of Staking account which holds staking balance
+	CouncilRewardAddrs  []common.Address // Address of Council account which will get block reward
+	KIRAddr             common.Address   // Address of KIR contract
+	PoCAddr             common.Address   // Address of PoC contract
+
+	UseGini bool
+	Gini    float64 // gini coefficient
+
+	// Derived from CouncilStakingAddrs
+	CouncilStakingAmounts []uint64 // Staking amounts of Council
+}
+
+func newEmptyStakingInfo(blockNum uint64) *StakingInfo {
```

Comment:
Is it a test-only function? 
Why don't you move the function to a test file?

Reply:
no. it will be used in the code